src/Features/Core/Portable/TodoComments/TodoCommentDescriptor.cs

```diff
@@ -24,28 +26,18 @@ public TodoCommentDescriptor(string text, int priority)
             Priority = priority;
         }
 
-        public static ImmutableArray<TodoCommentDescriptor> Parse(string data)
+        public static ImmutableArray<TodoCommentDescriptor> Parse(ImmutableArray<string> items)
         {
-            if (string.IsNullOrWhiteSpace(data))
-                return ImmutableArray<TodoCommentDescriptor>.Empty;
-
-            var tuples = data.Split('|');
             using var _ = ArrayBuilder<TodoCommentDescriptor>.GetInstance(out var result);
```

Comment:
Nit, but we can we pass capacity? (This probaby runs rarely enough to matter though.)

Reply:
(Feel free to ignore unless you have to make some other edit.)